### ISSUE TITLE
KeyboardDatePicker does not show errors

### ISSUE THREAD

**Issue by DatChoob** [stackworx/formik-material-ui#153]
I have two KeyboardDatePickers, a start date and an end date.

I am adding validation to show an error to the startDate if startDate > endDate.
However, this error does not show on the page.

When displaying errors on the render, I do not see startDate in the error object.

I have created an example here: https://codesandbox.io/s/hardcore-shirley-7m6fw

**Comment by hurricane987** [comment]
This is also happening with DatePicker

**Comment by nick-romano-kci** [comment]
> This is also happening with DatePicker

@hurricane987 if you need to fix this within datepicker, you can do the following implementation.

```
function fieldToDatePicker({ field, form: { isSubmitting, touched, errors, setFieldValue, setFieldError }, disabled, ...props }) {
  const fieldError = getIn(errors, field.name);
  const showError = getIn(touched, field.name) && !!fieldError;
  return {
      ...props,
      ...field,
      error: showError,
      helperText: showError ? fieldError : props.helperText,
      disabled: disabled != undefined ? disabled : isSubmitting,
      onChange(date) {
          console.log(date);
          setFieldValue(field.name, date);
      }
  };
}

const DateFormik = (props) => {
  return <DatePicker {...fieldToDatePicker(props)} ></DatePicker>
}
```

There's some bad code in the source that overrides the onError functions and creates the issues.

**Comment by dieguezz** [comment]
happening to me as well

**Comment by johnmagbag1995** [comment]
@nick-romano-kci how to implement this using https://codesandbox.io/s/hardcore-shirley-7m6fw

**Comment by dieguezz** [comment]
For the moment i am using a modified version of `KeyboardDatePicker.tsx` that makes the job. 

```
import * as React from 'react'
import {
  KeyboardDatePicker as MuiKeyboardDatePicker,
  KeyboardDatePickerProps as MuiKeyboardDatePickerProps,
} from '@material-ui/pickers'
import { FieldProps, getIn } from 'formik'

export interface KeyboardDatePickerProps
  extends FieldProps,
    Omit<MuiKeyboardDatePickerProps, 'name' | 'value' | 'error' | 'onChange'> {}

export function fieldToKeyboardDatePicker({
  disabled,
  field,
  form: { isSubmitting, touched, errors, status = {}, setFieldValue, setFieldError },
  ...props
}: KeyboardDatePickerProps): MuiKeyboardDatePickerProps {
  const fieldError = getIn(errors, field.name) || getIn(status.errors, field.name)

  return {
    ...props,
    ...field,
    error: !!fieldError,
    helperText: fieldError ?? props.helperText,
    disabled: disabled !== undefined ? disabled : isSubmitting,
    onChange(date) {
      setFieldValue(field.name, date)
    },
    onError(error) {
      if (error !== fieldError && !(error === '' && !fieldError)) {
        // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
        // @ts-ignore - https://github.com/jaredpalmer/formik/pull/2286
        if (error) {
            setFieldError(field.name, String(error))
        }
      }
    },
  }
}

export function KeyboardDatePicker({ children, ...props }: KeyboardDatePickerProps) {
  return (
    <MuiKeyboardDatePicker {...fieldToKeyboardDatePicker(props)}>{children}</MuiKeyboardDatePicker>
  )
}
```

**Comment by charlielow** [comment]
I'm seeing something similar with TimePicker as well, if form-level validation returns an error on a TimePicker field, those errors are not available on the errors object and don't affect properties related to validity

**Comment by cliedeman** [comment]
This is fixed in formik-material-ui-pickers@0.0.9

https://codesandbox.io/s/prod-grass-w9do3
